analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_snomed_6m(dx_codelist):
+  return patients.with_these_clinical_events(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 6 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_snomed_12m(dx_codelist):
+  return patients.with_these_clinical_events(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_icd(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning = "binary_flag",
+      on_or_before = "covid_test_positive_date",
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_icd_12m(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_ops(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_procedures=dx_codelist,
+      returning = "binary_flag",
+      on_or_before = "covid_test_positive_date",
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+          "rate": "exponential_increase",
+      },
+  )
+def comorbidity_ops_12m(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_procedures=dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.01,
+          "rate": "exponential_increase",
+      },
+  )
+def drug_6m(dx_codelist):
+  return patients.with_these_medications(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 6 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  )
+def drug_12m(dx_codelist):
+  return patients.with_these_medications(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  )
+def high_cost_drugs_3m(dx_codelist):
+  return patients.with_high_cost_drugs(
```

Comment:
High costs drugs data is only available between April 2018 and March 2020 so unfortunately this won't pick up what you want. This applies for all subsequent high risk drug variables throughout this study def.

Reply:
Hmm, I did actually know this. It does mean that we miss IMID patients on these drugs (which is our patients). I will remove this from the code